### PR TITLE
docs: fix types in PHPDocs in RESTful controllers

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See #6310

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/RESTful/ResourceController.php [hunk 1]
```diff
@@ -23,7 +24,7 @@ class ResourceController extends BaseResource
     /**
      * Return an array of resource objects, themselves in array format
      *
-     * @return mixed
+     * @return Response
```
Comment (MGatner) [inline system/RESTful/ResourceController.php:27]: All of the RESTful methods are placeholders to be extended (optionally) so even though their current return value is `Response` they could also return `string` or possibly even nothing. If we type these here without including those then developers will need to retype the docblocks or risk breaking static analysis. 

My suggestion is to make all these `Response|string|null`.

File: system/RESTful/ResourceController.php [hunk 2]
```diff
@@ -33,9 +34,9 @@ public function index()
     /**
      * Return the properties of a resource object
      *
-     * @param mixed $id
+     * @param string|null $id
```
Comment (MGatner) [inline system/RESTful/ResourceController.php:37]: I'm always torn on Controller parameters. The way you have it is probably the technically-correct solution, but lots of times resource IDs end up coming in as `int` during testing. This change would require casting test values.

**Review by MGatner — CHANGES_REQUESTED** [review]
Good changes, some points for discussion.

File: system/RESTful/ResourceController.php (hunk 1 above)
Comment (kenjis) [reply system/RESTful/ResourceController.php]: Changed to `Response|string|void`.

File: system/RESTful/ResourceController.php (hunk 2 above)
Comment (kenjis) [reply system/RESTful/ResourceController.php]: Changed to `int|string|null`.
We should support int type.

**Review by MGatner — APPROVED** [review]
🚀

**Pull request merged**